glean-core/src/storage/mod.rs

```diff
@@ -31,14 +31,30 @@ impl StorageManager {
         store_name: &str,
         clear_store: bool,
     ) -> Option<JsonValue> {
-        let mut snapshot: HashMap<&str, HashMap<String, JsonValue>> = HashMap::new();
+        let mut snapshot: HashMap<String, HashMap<String, JsonValue>> = HashMap::new();
 
         let mut snapshotter = |metric_name: &[u8], metric: &Metric| {
-            let map = snapshot
-                .entry(metric.category())
-                .or_insert_with(HashMap::new);
             let metric_name = String::from_utf8_lossy(metric_name).into_owned();
-            map.insert(metric_name, metric.as_json());
+            if metric_name.contains("/") {
+                let category = format!("labeled_{}", metric.category());
+                let map = snapshot.entry(category).or_insert_with(HashMap::new);
+
+                let mut s = metric_name.splitn(2, '/');
```

Comment:
MICROOPTIMIZATION TO FREELY IGNORE: Maybe there's a way to do this above so we don't have to scan through the string twice?